### PR TITLE
vscode-extensions: move underscore-prefixed extension identifiers to quoted ones

### PR DESCRIPTION
**Pull request by schuelermine**
###### Description of changes

Moved `_1Password` to `1Password`, `_2gua` to `2gua`, and `_4ops` to `4ops`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -31,7 +31,11 @@ let
   #
   baseExtensions = self: lib.mapAttrs (_n: lib.recurseIntoAttrs)
     {
-      _1Password.op-vscode = buildVscodeMarketplaceExtension {
+      _1Password = builtins.trace ''_1Password has been replaced with "1Password"'' self."1Password";
+      _2gua = builtins.trace ''_2gua has been replaced with "2gua"'' self."2gua";
+      _4ops = builtins.trace ''_4ops has been replaced with "4ops"'' self."4ops";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:36]:
```suggestion
      _1Password = lib.trace ''_1Password has been replaced with "1Password"'' self."1Password";
      _2gua = lib.trace ''_2gua has been replaced with "2gua"'' self."2gua";
      _4ops = lib.trace ''_4ops has been replaced with "4ops"'' self."4ops";
```
lib.trace adds some extra stuff which we should use https://github.com/NixOS/nixpkgs/blob/master/lib/sources.nix#L110-L131

Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix]: done

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -31,7 +31,11 @@ let
   #
   baseExtensions = self: lib.mapAttrs (_n: lib.recurseIntoAttrs)
     {
-      _1Password.op-vscode = buildVscodeMarketplaceExtension {
+      _1Password = lib.trace ''_1Password has been replaced with "1Password"'' self."1Password";
+      _2gua = lib.trace ''_2gua has been replaced with "2gua"'' self."2gua";
+      _4ops = lib.trace ''_4ops has been replaced with "4ops"'' self."4ops";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:36]: This needs to be guarded behind config.allowAliases == false.

Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix:36]: Done!

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -2750,7 +2754,7 @@ let
           license = lib.licenses.mit;
         };
       };
-    };
+    });
 
   aliases = self: super: {
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/editors/vscode/extensions/default.nix:2759]: We should merge both of the alias functions preferable keeping the old and adding the new aliases to it.

**Comment by ShamrockLee** [comment]
What's the purpose of this change?

I think the underscore prefix is to enable us to call the attribute without having to quote it.

**Comment by schuelermine** [comment]
Having to quote these names is not a great inconvenience. It’s an intended feature of the Nix language, and I think we should use it more often. These attribute paths are based on the unique identifiers used for the extensions in the Visual Studio Marketplace, and using anything different risks confusing users. The fact that autocomplete does not work for them is an issue with `nix repl` that ought to be fixed and not a reason to contort the language.

**Comment by ShamrockLee** [comment]
That makes sense. It would also simplify the relation between the VSCode Marketplace registry and our attrset.

BTW, is it possible to use digit-prefixed attributes inside the function input?

**Comment by schuelermine** [comment]
hm, I don’t think it is. You can do `args@{...}` though.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -2750,7 +2754,7 @@ let
...
     # aliases
```
Comment (ShamrockLee) [inline pkgs/applications/editors/vscode/extensions/default.nix:2760]: @schuelermine  Sandro probably means

```suggestion
    # aliases
    _1Password = lib.trace ''_1Password has been replaced with "1Password"'' self."1Password";
    _2gua = lib.trace ''_2gua has been replaced with "2gua"'' self."2gua";
    _4ops = lib.trace ''_4ops has been replaced with "4ops"'' self."4ops";
```

and remove the new alias section at line 33-37.

**Review by ShamrockLee — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 4 above)
Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix:2756]: Ah, I see.
I’ve done so now.

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix:2755]: I understand now. I have done so

**Comment by superherointj** [comment]
Please fix merge conflict.

**Comment by schuelermine** [comment]
@superherointj I have!

**Comment by superherointj** [comment]
Eval failed. Please check. You can eval locally as well.

**Comment by schuelermine** [comment]
The ofborg-eval failure is just due to “ This PR does not cleanly list package outputs after merging. ”, and not because the packages don’t evaluate.

Screenshot from 2023-03-18 23-59-09

**Comment by superherointj** [comment]
> The ofborg-eval failure is just due to “ This PR does not cleanly list package outputs after merging. ”, and not because the packages don’t evaluate.
> 
> Screenshot from 2023-03-18 23-59-09
> 

Right. Sorry. When I am back to a PC, I will see this PR. Phone is a bit bad.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -32,8 +32,8 @@ let
   # "${mktplcRef.publisher}.${mktplcRef.name}".
   #
   baseExtensions = self: lib.mapAttrs (_n: lib.recurseIntoAttrs)
-    {
-      _1Password.op-vscode = buildVscodeMarketplaceExtension {
+    ({
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:35]:
```suggestion
    ({
```
Why `(` is necessary here?

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 6]
```diff
@@ -3177,13 +3177,15 @@ let
           license = lib.licenses.mit;
         };
       };
-
-    };
+    });
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3180]:
```suggestion
    };
```
Is this really necessary?

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 5 above)
Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix:35]: it isn’t, thanks for noticing

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 6 above)
Comment (schuelermine) [reply pkgs/applications/editors/vscode/extensions/default.nix:3180]: removed it

**Comment by superherointj** [comment]
Please squash commits.

**Comment by superherointj** [comment]
Hi schuelermine.
It should be a single commit.
Try to (re-)fetch upstream. Then, rebase to master.

**Comment by superherointj** [comment]
ping @schuelermine

**Comment by schuelermine** [comment]
@superherointj I seem to have messed up my git repository. I will extract the changes and make a new PR on a different branch.

**Comment by schuelermine** [comment]
This change is now at #222091

**Comment by superherointj** [comment]
> The ofborg-eval failure is just due to “ This PR does not cleanly list package outputs after merging. ”, and not because the packages don’t evaluate.
> 
> Screenshot from 2023-03-18 23-59-09

Now we better understand this error breaks with CI. Eval was OK. But 'cleanly outputs' was not.

**Pull request closed without merging**